trieste/models/model_interfaces.py

```diff
@@ -60,110 +70,19 @@ def sample(self, query_points: QueryPoints, num_samples: int) -> ObserverEvaluat
         """
         raise NotImplementedError
 
-    @abstractmethod
-    def optimize(self) -> None:
-        """ Optimize the model parameters. """
-        raise NotImplementedError
-
-
-Optimizer = Union[gpflow.optimizers.Scipy, tf.optimizers.Optimizer]
-
-
-class TrainableModelInterface(tf.Module, ModelInterface, ABC):
-    """
-    A utility class that provides a default optimization strategy, as well as the ability to modify
-    various elements of this strategy.
 
-    :cvar maxiter_default: The default maximum number of iterations to optimize the model for, when
-        using a :class:`tf.optimizers.Optimizer`.
-    :cvar create_optimizer_default: Builder for the default optimizer.
-    :cvar apply_jit_function: If `True`, the default optimization procedure is compiled
-        with :func:`tf.function`.
-    """
-
-    maxiter_default = 1000
-    create_optimizer_default: Callable[[], Optimizer] = gpflow.optimizers.Scipy
-    apply_jit_function = False
+class GPflowPredictor(ModelInterface, ABC):
+    """ A trainable wrapper for a GPflow Gaussian process model. """
 
-    @abstractmethod
-    def loss(self) -> tf.Tensor:
-        """ The training loss (to be minimized) on this model. """
-        raise NotImplementedError
+    def __init__(self, optimizer: Optional[Optimizer] = None):
+        if optimizer is None:
+            optimizer = Optimizer(gpflow.optimizers.Scipy())
+        self._optimizer = optimizer
```

Comment:
can this be on a module-private subclass of `GPflowPredictor`? It makes it easier to define implementations of this interface